pkg/ccl/backupccl/restore_span_covering_test.go

```diff
@@ -504,3 +505,64 @@ func TestRestoreEntryCover(t *testing.T) {
 		}
 	}
 }
+
+// TestFilterCompletedImportSpans checks that all completed spans are
+// skipped over when creating the completed span frontier and toDoSpans slice in memory.
+func TestFilterCompletedImportSpans(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	sp := func(start, end string) roachpb.Span {
+		return roachpb.Span{Key: roachpb.Key(start), EndKey: roachpb.Key(end)}
+	}
+	f := func(start, end, path string) backuppb.BackupManifest_File {
+		return backuppb.BackupManifest_File{Span: sp(start, end), Path: path}
+	}
+	paths := func(names ...string) []execinfrapb.RestoreFileSpec {
+		r := make([]execinfrapb.RestoreFileSpec, len(names))
+		for i := range names {
+			r[i].Path = names[i]
+		}
+		return r
+	}
+
+	// Setup and test the example in the comment of makeSimpleImportSpans.
+	spans := []roachpb.Span{sp("a", "f"), sp("f", "i"), sp("l", "m")}
+	backups := []backuppb.BackupManifest{
+		{Files: []backuppb.BackupManifest_File{f("a", "c", "1"), f("c", "e", "2"), f("h", "i", "3")}},
+		{Files: []backuppb.BackupManifest_File{f("b", "d", "4"), f("g", "i", "5")}},
+		{Files: []backuppb.BackupManifest_File{f("a", "h", "6"), f("j", "k", "7")}},
+		{Files: []backuppb.BackupManifest_File{f("h", "i", "8"), f("l", "m", "9")}},
+	}
+
+	emptySpanFrontier, err := spanUtils.MakeFrontier(roachpb.Span{})
+	require.NoError(t, err)
+
+	importSpans := makeSimpleImportSpans(spans, backups, nil, emptySpanFrontier, nil, 2<<20)
+
+	// Tests that no completed spans returns a frontier with all empty timestamps and the original import span slice.
+	completedImportSpans, newImportSpans, err := filterCompletedImportSpans(nil, importSpans)
```

Comment:
this approach looks correct, though I think we could get better coverage with less code by creating a helper test function that gets called in `TestRestoreEntryCoverExample` with the following signature:

```
checkSpanFiltering simulates and checks the filterCompletedImportSpans function by filtering the expectedCompletedSpans from the provided ImportSpans and checks the expected todoSpans after filtering
checkSpanFiltering(expectedCompletedSpans,expectedToDoSpans,importSpans)
```

This means we don't need a brand new unit test here.

Reply:
should this function be declared within TestRestoreEntryCoverExample?

Reply:
i don't think it needs to be, so I wouldn't. In general, don't use closures unless you need to.